.ci/build_docs.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+set -xueo pipefail
+
+# Check that documentation was generated successfully
+if [[ "$GHC_VERSION" = "8.6.5" ]]; then
+  haddock_pkgs="clash-lib clash-cosim"
+else
+  haddock_pkgs="clash-prelude clash-lib clash-cosim"
+fi
+
+mkdir -p hadocs
+
+for pkg in ${haddock_pkgs}; do
+  set -e
+
+  # Cache dependencies
+  cabal v2-build ${pkg} -O0 --enable-documentation --only-dependencies
+
+  # HaskellPrelude yields warnings we cannot fix
+  cabal v2-haddock ${pkg} -O0 --enable-documentation \
+    |& grep -v "Clash.HaskellPrelude" \
+    |  tee haddock_log
+
+  set +e
+
+  # TODO: We're not yet ready for the check below :-)
+  # if grep -q "Missing documentation" haddock_log; then
+  #   echo -e "\e[1m\e[31mMissing documentation! Scroll up for full log.\e[0m"
+  #   grep --color=always -n -C 5 "Missing documentation" haddock_log
+  #   exit 1
+  # fi
+
+  if [[ $pkg == "clash-prelude" ]]; then
+    # Currently just checking clash-prelude; other libraries still fail these
+    # checks.
+
+    out_of_scope_warn="If you qualify the identifier, haddock can try to link it anyway"
+    if grep -q "${out_of_scope_warn}" haddock_log; then
+      echo -e "\e[1m\e[31mIdentifier out of scope in ${pkg}! Scroll up for full log.\e[0m"
+      grep --color=always -n -C 5 "${out_of_scope_warn}" haddock_log
```

Comment:
I would test that ANSI colour is supported, and only use colour if it is. You can do

```bash
test -t 1 # check if stdout is a terminal
tput colors # outputs the number of supported colors, use ANSI if == 256
```

this way if you run locally and pipe the output to a file you don't get a bunch of ANSI escapes that a text editor won't interpret correctly

Reply:
This also means using the default `--color=auto` for grep etc.